doc/error-handling.md

```diff
@@ -0,0 +1,323 @@
+This could also be done in RouDi (by sending information to RouDi), but RouDi already has too much responsibility. Preferably this should be done by a separate application with this sole purpose.
+If the application cannot reach the central handler, it shall try to handle the error locally if possible (at least log it).
+
+However, it might be too slow if this would rely on error transmission and responses. If this is to be implemented, the exact mechanism has to be decided on.
+
+## 3rd Party Error Handling
+We need to decide how to provide an interface for 3rd party error handling, especially for the runtime. This interface will probably rely on hooks/callbacks. The signature and callsites of these need to be discussed.
+This is related to centralized error handling as well.
+
+## Overriding Specific Error Reaction
+* Do we want to provide the ability to override error reaction based on e.g. error codes?
+* Do we want to disable certain error levels? (if so, this should preferably happen at compile 
+time with no or few runtime artifacts). It could be an option to e.g. disable all ``MODERATE`` error reaction at compile time.
+* It is probably not reasonable to allow disabling reaction on fatal errors.
+
+This is also related to the hooks for 3rd party error handling we may want to provide.
+
+## Return in Case of Fatal Error
+The reporting code does not need to be able to continue properly in case of a fatal error, but there needs to be a return after the error handler call. While the error handler is not required to return, it still might under certain circumstances (e.g. a mock error handler in a test case). 
+
+The (complete) intended behavior of the error handler requires some further clarification, especially in the case of fatal errors. In the case of non-fatal errors the code invoking the error handler must be able to continue after the error handler returns.
+
+## Error Handling vs. Logging
+Does it make sense to have ``LogErr`` without an error handler call? If an error occurs it should probably be enforced that the handler is called and not just lead to a log entry.
+One reason for this is that currently it is not possible to provide an additional error message to the error handler.
+
+## Additional Error Information
+It would be desirable to allow the possibility to provide additional messages (or even general functions/arguments) to the error handler, which is currently missing.
+This can be combined with the addition of error location to the error handler.
+
+An optional stack-trace (at least in debug mode) may also prove very useful. 
+What is needed to have a limited stack-trace even in release mode?
+
+## Debug vs. release mode
+We need to further clarify behavior in Release and debug mode of the error handler and ``Expects`` and ``Ensures`` (and maybe the logger as well). Can we have a release build with additional information? (e.g. symbols for a stack-trace).
```

Comment:
release (lowercase initial) for consistency :smile:

Reply:
Consistency is overrated...